google/cloud/security/actions/action_config_validator.py

```diff
@@ -0,0 +1,202 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Actions configuration validator."""
+
+from collections import defaultdict
+import importlib
+import os
+import yaml
+
+class ActionsConfigValidator(object):
+    """ActionsConfigValidator class.
+
+    Contains basic validation that should be done on all action configurations.
+    """
+
+    @staticmethod
+    def validate(actions_config_path):
+        """Validates the actions config.
+
+        Action configuration validation:
+        1. Config is a valid yaml file.
+        2. Each action has an id and it is unique.
+        3. Check each action for:
+          1. An action type.
+          2. That the action type is valid.
+          3. A trigger.
+          4. That the trigger is valid.
+
+        Args:
+            actions_config_path (str): The actions configuration path.
+
+        Returns:
+            tuple: A tuple with the dict configuration and the list of errors.
+
+        Raises:
+            ConfigLoadError: If the config cannot be loaded.
+        """
+        errors = []
+        config = ActionsConfigValidator._load_and_validate_yaml(
+            actions_config_path)
+
+        actions, errs = ActionsConfigValidator._load_actions(config)
+        errors.extend(errs)
+
+        for action in actions:
+            action_errs = []
+            action_errs.append(
+                ActionsConfigValidator._check_action_type(action))
+            action_errs.append(ActionsConfigValidator._check_trigger(action))
+            for err in action_errs:
+                if err is not None:
+                    errors.append(err)
+        return config, errors
+
+    @staticmethod
+    def _load_and_validate_yaml(path):
+        """Loads the configuration from a path.
+
+        Args:
+          path (str): The path the the configuration.
+
+        Returns:
+          dict: The loaded configuration dictionary.
+
+        Raises:
+          ConfigLoadError: If the config cannot be loaded.
+        """
+        with open(os.path.abspath(path), 'rb') as filep:
+            try:
+                config = yaml.safe_load(filep)
+            except ValueError as yaml_error:
+                raise ConfigLoadError(yaml_error)
+        return config
+
+    @staticmethod
+    def _load_actions(config):
+        """Loads the actions actions from the config.
+
+        Args:
+          config (dict): A dictionary configuration.
+
+        Returns:
+          Tuple: A tuple of the actions and errors.
+        """
+        action_ids = defaultdict(int)
+        actions = []
+        errors = []
+        for action in config.get('actions', []):
+            action_id = action.get('id')
+            if action_id is None:
+                errors.append(MissingRequiredActionField('id'))
+            action_ids[action_id] += 1
```

Comment:
I recommend making action_ids a set and then you change check if action_id in action_ids: add error.